code/modules/core_implant/cruciform/machinery/eotp.dm

```diff
@@ -0,0 +1,196 @@
+	..()
+	eotp = src
+	START_PROCESSING(SSobj, src)
+
+/obj/machinery/power/eotp/examine(user)
+	..()
+
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		var/obj/item/weapon/implant/core_implant/I = H.get_core_implant(/obj/item/weapon/implant/core_implant/cruciform)
+		if(I && I.active && I.wearer)
+			var/comment = "Power level: [power]/[max_power]."
+			comment += "\nObservation level: [observation]/[max_observation]."
+			to_chat(user, SPAN_NOTICE(comment))
+
+/obj/machinery/power/eotp/Process()
+	..()
+	if(stat)
+		return
+
+	updateObservation()
+
+	if(world.time >= (last_rescan + rescan_cooldown))
+		var/mob/living/carbon/human/H = pick(scanned)
+		var/obj/item/weapon/implant/core_implant/I = H.get_core_implant(/obj/item/weapon/implant/core_implant/cruciform)
+		if(I && I.active && I.wearer)
+			eotp.removeObservation(20)
+		else if (is_carrion(H))
+			eotp.addObservation(20)
+		else
+			eotp.removeObservation(10)
+
+		scanned.Remove(H)
+		last_rescan = world.time
+
+	updatePower()
+
+/obj/machinery/power/eotp/proc/addObservation(var/number)
+	observation += number
+	return observation
+
+/obj/machinery/power/eotp/proc/removeObservation(var/number)
+	observation -= number
+	return observation
+
+/obj/machinery/power/eotp/proc/updateObservation()
+	if(observation > max_observation)
+		observation = max_observation
+
+	if(observation < min_observation)
+		observation = min_observation
+
+/obj/machinery/power/eotp/proc/updatePower()
+	power_gaine = initial(power_gaine) + (observation / 100)
+
+	if(world.time >= (last_power_update + power_cooldown))
+		power += power_gaine
+		last_power_update = world.time
+
+	if(power >= max_power)
+		power -= max_power
+		power_release()
+
+/obj/machinery/power/eotp/proc/power_release()
+	var/type_release = pick(rewards)
+
+	if(type_release == ARMAMENTS)
+		var/reward_disk = pick(disk_types)
+		var/obj/item/_item = new reward_disk(get_turf(src))
+		visible_message(SPAN_NOTICE("The [_item.name] appers out of bluespace near the [src]!"))
+
+	else if(type_release == ALERT)
+
+		var/area/antagonist_area
+		var/preacher
+
+		for(var/datum/antagonist/A in GLOB.current_antags)
+			if((A.id == ROLE_CARRION) ||(A.id == ROLE_BLITZ) || (A.id == ROLE_BORER))
+				var/mob/living/L = A.owner.current
+				if(!isghost(L))
+					antagonist_area = get_area(L)
+					break
+		if(!antagonist_area)
+			for(var/disciple in disciples)
+				to_chat(disciple, SPAN_NOTICE("You can sleep safely. God protects you with his eye."))
+				if(ishuman(disciple))
+					var/mob/living/carbon/human/H = disciple
+					if(H.sanity)
+						H.sanity.changeLevel(20)
+			return
+
+		for(var/disciple in disciples)
+			if(ishuman(disciple))
+				var/mob/living/carbon/human/H = disciple
+				if(H.mind)
+					var/assigned_job = H.mind.assigned_job
+					if(istype(assigned_job, /datum/job/chaplain))
+						preacher = H
+
+		if(!preacher)
+			preacher = pick(disciples)
+
+		to_chat(preacher, SPAN_DANGER("You feel the evil in [antagonist_area]!"))
```

Comment:
engrish

```suggestion
		to_chat(preacher, SPAN_DANGER("You feel an evil presence lurking in the [antagonist_area].")) // will say 'you feel an evil presence lurking in the Kitchen' or whatever
```

Reply:
I will remove before the [antagonist_area]. [antagonist_area] already adds :the:. (Like "the Bar" or others.)